flair/embeddings.py

```diff
@@ -477,49 +475,66 @@ def embed(self, paragraphs: Union[List[Sentence], Sentence]):
+                 use_first_representation: bool = False):
         """The constructor takes a list of embeddings to be combined.
-        :param word_embeddings: a list of word embeddings
+        :param token_embeddings: a list of token embeddings
         :param hidden_states: the number of hidden states in the lstm
         :param num_layers: the number of layers for the lstm
         :param reproject_words: boolean value, indicating whether to reproject the word embedding in a separate linear
         layer before putting them into the lstm or not
+        :param reproject_words_dimension: output dimension of reprojecting words. If None the same output dimension as
+        before will be taken.
         :param bidirectional: boolean value, indicating whether to use a bidirectional lstm or not
+        :param use_first_representation: boolean value, indicating whether to concatenate the first and last
+        representation of the lstm to be used as final document embedding.
         """
         super().__init__()
 
-        self.embeddings: List[TokenEmbeddings] = word_embeddings
+        self.embeddings: List[TokenEmbeddings] = token_embeddings
 
         self.reproject_words = reproject_words
         self.bidirectional = bidirectional
+        self.use_first_representation = use_first_representation
 
-        self.length_of_all_word_embeddings = 0
-        for word_embedding in self.embeddings:
-            self.length_of_all_word_embeddings += word_embedding.embedding_length
+        self.length_of_all_token_embeddings = 0
+        for token_embedding in self.embeddings:
+            self.length_of_all_token_embeddings += token_embedding.embedding_length
 
-        self.name = 'text_lstm'
+        self.name = 'document_lstm'
         self.static_embeddings = False
 
+        self.__embedding_length: int = hidden_states
         if self.bidirectional:
-            self.__embedding_length: int = hidden_states * 2
-        else:
-            self.__embedding_length: int = hidden_states
+            self.__embedding_length *= 2
+        if self.use_first_representation:
+            self.__embedding_length *= 2
+
+        self.embeddings_dimension: int = self.length_of_all_token_embeddings
+        if self.reproject_words and reproject_words_dimension is not None:
+            self.embeddings_dimension = reproject_words_dimension
 
         # bidirectional LSTM on top of embedding layer
-        self.word_reprojection_map = torch.nn.Linear(self.length_of_all_word_embeddings,
-                                                     self.length_of_all_word_embeddings)
-        self.rnn = torch.nn.LSTM(self.length_of_all_word_embeddings, hidden_states, num_layers=num_layers,
+        self.word_reprojection_map = torch.nn.Linear(self.length_of_all_token_embeddings,
+                                                     self.embeddings_dimension)
+        self.rnn = torch.nn.GRU(self.embeddings_dimension, hidden_states, num_layers=num_layers,
```

Comment:
why GRU and not LSTM?

Reply:
During training GRU worked slightly better than LSTMs on the small datasets.

Reply:
:+1: